jobapi/middleware.go

```diff
@@ -0,0 +1,47 @@
+package jobapi
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+)
+
+func AuthMdlw(token string) func(http.Handler) http.Handler {
```

Comment:
I don't want to nitpick about naming and doc comments, but also I very strongly *want* to nitpick about naming and doc comments...

Reply:
![doit](https://user-images.githubusercontent.com/15753101/217404614-487cfd8e-9342-4f42-ab76-e335964b6b48.gif)

Reply:
Alright. 😈

Could you 💐please💐 add (throughout) doc comments to the top-level exported identifiers. This doesn't need to happen right away, but ideally before the PR is ready to go.

```go
// AuthMdlw returns a middleware that checks for the given bearer token
// before invoking the next handler.
```

I'm also wondering why `Mdlw` is abbreviated... I suppose I can understand if you say vowels are expensive?

Reply:
> I'm also wondering why Mdlw is abbreviated... I suppose I can understand if you say vowels are expensive?
you know, i don't know. i thiiiiiiiiiink that it comes from some http library that i used yonks ago that called its middleware package `mdlw`, but it's a bad reason to abbreviate 😅